src/create-client.ts

```diff
@@ -0,0 +1,42 @@
+import { IProxyConfig } from './config';
+import Client from './client';
+import { initialize } from 'unleash-client';
+import Metrics from 'unleash-client/lib/metrics';
+import { defaultStrategies } from 'unleash-client/lib/strategy';
+
+export const createClient = (config: IProxyConfig): Client => {
```

Comment:
we moved the actual assembly of dependencies to this new file that owns this responsibility making client much simpler and more testable

Reply:
This is so-called composition root: https://blog.ploeh.dk/2011/07/28/CompositionRoot/